core/net/rpl/rpl-ext-header.c

```diff
@@ -378,20 +388,20 @@ insert_srh_header()
     cmpri = MIN(cmpri, count_matching_bytes(&node_addr, &UIP_IP_BUF->destipaddr, 16));
     cmpre = cmpri;
 
-    PRINTF("RPL: SRH Hop ");
-    PRINT6ADDR(&node_addr);
-    PRINTF("\n");
+    if(node->parent != root_node) {
+      PRINTF("RPL: SRH Hop ");
+      PRINT6ADDR(&node_addr);
+      PRINTF("\n");
+    } else if(((DEBUG) & DEBUG_PRINT)) {
+      rpl_ns_get_node_global_addr(&node_addr, node);
+      PRINTF("RPL: SRH Next Hop ");
+      PRINT6ADDR(&node_addr);
+      PRINTF("\n");
+    }
     node = node->parent;
     path_len++;
   }
 
-  if(((DEBUG) & DEBUG_PRINT) && node != NULL) {
-    rpl_ns_get_node_global_addr(&node_addr, node);
-    PRINTF("RPL: SRH Next Hop ");
-    PRINT6ADDR(&node_addr);
-    PRINTF("\n");
-  }
```

Comment:
Oh, I overlooked `rpl_ns_get_node_global_addr` staying at line-385! Which do you mean by `the whole block`, only the `else if` block or the entire `if ... else if ...` block? Either choice would be okay for me, anyway.